message-hub/README.md

```diff
@@ -0,0 +1,174 @@
+producer.ssl.protocol=TLSv1.2
+producer.ssl.enabled.protocols=TLSv1.2
+producer.ssl.endpoint.identification.algorithm=HTTPS
+
+# Message consumer connection configuration, substitute your username and password
+consumer.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="username" password="password";
+consumer.security.protocol=SASL_SSL
+consumer.sasl.mechanism=PLAIN
+consumer.ssl.protocol=TLSv1.2
+consumer.ssl.enabled.protocols=TLSv1.2
+consumer.ssl.endpoint.identification.algorithm=HTTPS
+
+# Substitue with the path to the file you want to use to store offset values
+offset.storage.file.filename=offsetFile
+
+# Kafka connect message configuration
+key.converter=org.apache.kafka.connect.json.JsonConverter
+value.converter=org.apache.kafka.connect.json.JsonConverter
+key.converter.schemas.enable=true
+value.converter.schemas.enable=true
+internal.key.converter=org.apache.kafka.connect.json.JsonConverter
+internal.value.converter=org.apache.kafka.connect.json.JsonConverter
+```
+
+# Configuration for publishing messages from Cloudant to Message Hub using Apache Kafka and kafka-connect-cloudant
+
+1. Download the [`kafka-connect-cloudant` jar from Maven central](TODO insert link when available) - this tutorial used `kafka-connect-cloudant-0.100.0-kafka-1.0.0.jar`.
+1. Create a `connect-cloudant-source.properties` file.
+1. Add these properties, substituting your details where appropriate.
+
+```properties
+# Substitute with a name for your source connector instance
+name=cdt-kafka-example
+connector.class=com.ibm.cloudant.kafka.connect.CloudantSourceConnector
+# Substitute with the name of your topic
+topics=exampleTopic
+# Substitute with your DB
+cloudant.db.url=https://examples.cloudant.com/animaldb
+# Not needed for the public examples account, but if using your own data then
+# substitute with your Cloudant credentials
+#cloudant.db.username=
+#cloudant.db.password=
+cloudant.db.since=0
+batch.size=50
+# Configure kafka-connect-cloudant to generate structs from the JSON documents
+cloudant.value.schema.struct=true
+# Flattening is not needed for this tutorial, but if you have nested objects and
+# arrays in your JSON documents consider enabling flattening for use with the
+# kafka-connect-jdbc sink. This may not be necessary for other sink connectors
+# that can handle nested structs.
+cloudant.value.schema.struct.flatten=false
+# Configure kafka-connect-cloudant to omit design documents
+cloudant.omit.design.docs=true
+```
+
+# Configuration for consuming messages from Message Hub and writing to Db2 Warehouse on Cloud using Apache Kafka's integrated jdbc-sink-connector
+
+1. Download the [`kafka-connect-jdbc` jar from Confluent's Maven repository](packages.confluent.io/maven/io/confluent/kafka-connect-jdbc/4.0.0/kafka-connect-jdbc-4.0.0.jar)
+1. Create a `connect-jdbc-sink.properties` file
+1. Add these properties, substituting your details where appropriate.
+
+```properties
```

Comment:
You can indent lines 129 to 139 to match up with the indent of step 3.

Reply:
Done in 60970d1ba8e31da144cc94261b4b4c7867092d90